src/librepgp/validate.c

```diff
@@ -797,7 +676,7 @@ pgp_validate_file(pgp_io_t *             io,
     int                infd;
     int                cc;
 
-    if (stat(infile, &st) < 0) {
+    if ((stat(infile, &st) < 0) && io) {
```

Comment:
This check is wrong

This function should return _ERROR_BAD_PARAMETERS if "io==NULL" instead.

What's the goal of printing errors to "io->errs" at all? Why not to simply errors to "stderr" ?

Reply:
See https://github.com/riboseinc/rnp/issues/157.